plugin/log/setup.go

```diff
@@ -51,45 +54,57 @@ func logParse(c *caddy.Controller) ([]Rule, error) {
 		} else {
 			// Name scope, and maybe a format specified
 			format := DefaultLogFormat
+			last := len(args)
 
-			switch args[1] {
-			case "{common}":
-				format = CommonLogFormat
-			case "{combined}":
-				format = CombinedLogFormat
-			default:
-				format = args[1]
+			if strings.Contains(args[last-1], "{") {
+				last--
+				switch args[last] {
+				case "{common}":
+					format = CommonLogFormat
+				case "{combined}":
+					format = CombinedLogFormat
+				default:
+					format = args[last]
+				}
 			}
 
-			rules = append(rules, Rule{
-				NameScope: dns.Fqdn(args[0]),
-				Format:    format,
-				Class:     make(map[response.Class]struct{}),
-			})
+			args = args[:last]
```

Comment:
can we not just chopped of the last element in the `strings.Contains` bit above? Then `args` is always correct, and we don't need to last even.

Reply:
awesome !! i will take it~

Reply:
Hi,

I have make the changes, please have a check.

Thank you !!!